src/MSBuild/LiveLogger/Terminal.cs

```diff
@@ -29,18 +30,51 @@ internal sealed class Terminal : ITerminal
     /// </summary>
     private bool _isBuffering = false;
 
+    internal TextWriter Output { private get; set; } = Console.Out;
+
+    private const int BigUnknownDimension = 2 << 23;
+
     /// <inheritdoc/>
-    public int Height => Console.BufferHeight;
+    public int Height
+    {
+        get
+        {
+            if (Console.IsOutputRedirected)
```

Comment:
Just to be pedantic, you could theoretically redirect output to StdOut, right? (Not that that does anything) I'm wondering if there are ways to figure out the right dimensions in certain special cases. If you don't want to bother, I don't think it's actually important.

Reply:
I don't think that's interesting enough to jump through hoops for at this point.